src/Microsoft.ML.TimeSeries/DTRootCauseLocalization.cs

```diff
@@ -0,0 +1,306 @@
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | <xref: Microsoft.ML.Transforms.TimeSeries.RootCauseLocalizationInput> |
+    /// | Output column data type | <xref:System.Drawing.RootCause> |
+    /// | Exportable to ONNX | No |
+    ///
+    /// [!include[io](~/../docs/samples/docs/api-reference/time-series-root-cause-localization-dt.md)]
+    ///
+    /// The resulting <xref:Microsoft.ML.Transforms.Image.DTRootCauseLocalizationTransformer> creates a new column, named as specified in the output column name parameters, and
+    /// localize the root causes which contribute most to the anomaly.
+    /// Check the See Also section for links to usage examples.
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="TimeSeriesCatalog.LocalizeRootCauseByDT" />
+    public sealed class DTRootCauseLocalizationEstimator : TrivialEstimator<DTRootCauseLocalizationTransformer>
+    {
+        internal static class Defaults
+        {
+            public const double Beta = 0.5;
+        }
+
+        /// <summary>
+        /// Localize root cause.
+        /// </summary>
+        /// <param name="env">The estimator's local <see cref="IHostEnvironment"/>.</param>
+        /// <param name="beta">The weight for generating score in output result.</param>
+        /// <param name="columns">Pairs of columns to run the root cause localization.</param>
+        [BestFriend]
+        internal DTRootCauseLocalizationEstimator(IHostEnvironment env, double beta = Defaults.Beta, params (string outputColumnName, string inputColumnName)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(DTRootCauseLocalizationEstimator)), new DTRootCauseLocalizationTransformer(env, beta, columns))
+        {
+        }
+
+        /// <summary>
+        /// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
+        /// Used for schema propagation and verification in a pipeline.
+        /// </summary>
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.inputColumnName, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.inputColumnName);
+                if (!(col.ItemType is RootCauseLocalizationInputDataViewType))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.inputColumnName, new RootCauseLocalizationInputDataViewType().ToString(), col.GetTypeString());
+
+                result[colInfo.outputColumnName] = new SchemaShape.Column(colInfo.outputColumnName, col.Kind, col.ItemType, col.IsKey, col.Annotations);
```

Comment:
>col.Annotations [](start = 135, length = 15)

Doing this will pass the annotations from the input to the output. If this is not your intention then you do not need to pass this argument at all (it has a default value of `null`).

Reply:
> > col.Annotations [](start = 135, length = 15)
> 
> Doing this will pass the annotations from the input to the output. If this is not your intention then you do not need to pass this argument at all (it has a default value of `null`).

Updated to null